pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -719,3 +725,176 @@ func (k *kubeAPIServer) probeEndpoint(path string) string {
 }
+
+func (k *kubeAPIServer) computeKubeAPIServerCommand() []string {
+	var out []string
+
+	if versionutils.ConstraintK8sGreaterEqual117.Check(k.values.Version) {
+		out = append(out, "/usr/local/bin/kube-apiserver")
+	} else {
+		out = append(out, "/hyperkube", "kube-apiserver")
+	}
+
+	if k.values.VPN.ReversedVPNEnabled {
+		out = append(out, fmt.Sprintf("--egress-selector-config-file=%s/%s", volumeMountPathEgressSelector, configMapEgressSelectorDataKey))
+	}
+
+	out = append(out, "--enable-admission-plugins="+strings.Join(k.admissionPluginNames(), ","))
+	out = append(out, fmt.Sprintf("--admission-control-config-file=%s/%s", volumeMountPathAdmissionConfiguration, configMapAdmissionDataKey))
+	out = append(out, "--allow-privileged=true")
+	out = append(out, "--anonymous-auth="+strconv.FormatBool(k.values.AnonymousAuthenticationEnabled))
+	out = append(out, "--audit-log-path=/var/lib/audit.log")
+	out = append(out, fmt.Sprintf("--audit-policy-file=%s/%s", volumeMountPathAuditPolicy, configMapAuditPolicyDataKey))
+	out = append(out, "--audit-log-maxsize=100")
+	out = append(out, "--audit-log-maxbackup=5")
+	out = append(out, "--authorization-mode=Node,RBAC")
+
+	if k.values.SNI.Enabled {
+		out = append(out, fmt.Sprintf("--advertise-address=%s", k.values.SNI.AdvertiseAddress))
+	}
+
+	if len(k.values.APIAudiences) > 0 {
+		out = append(out, "--api-audiences="+strings.Join(k.values.APIAudiences, ","))
+	}
+
+	if k.values.BasicAuthenticationEnabled {
+		out = append(out, fmt.Sprintf("--basic-auth-file=%s/%s", volumeMountPathBasicAuthentication, secrets.DataKeyCSV))
+	}
+
+	out = append(out, fmt.Sprintf("--client-ca-file=%s/%s", volumeMountPathCA, secrets.DataKeyCertificateCA))
+	out = append(out, "--enable-aggregator-routing=true")
+	out = append(out, "--enable-bootstrap-token-auth=true")
+	out = append(out, "--http2-max-streams-per-connection=1000")
+	out = append(out, fmt.Sprintf("--etcd-cafile=%s/%s", volumeMountPathCAEtcd, secrets.DataKeyCertificateCA))
+	out = append(out, fmt.Sprintf("--etcd-certfile=%s/%s", volumeMountPathEtcdClient, secrets.DataKeyCertificate))
+	out = append(out, fmt.Sprintf("--etcd-keyfile=%s/%s", volumeMountPathEtcdClient, secrets.DataKeyPrivateKey))
+	out = append(out, fmt.Sprintf("--etcd-servers=https://%s:%d", etcd.ServiceName(v1beta1constants.ETCDRoleMain), etcd.PortEtcdClient))
+	out = append(out, fmt.Sprintf("--etcd-servers-overrides=/events#https://%s:%d", etcd.ServiceName(v1beta1constants.ETCDRoleEvents), etcd.PortEtcdClient))
+	out = append(out, fmt.Sprintf("--encryption-provider-config=%s/%s", volumeMountPathEtcdEncryptionConfig, SecretEtcdEncryptionConfigurationDataKey))
+	out = append(out, "--external-hostname="+k.values.ExternalHostname)
+
+	if k.values.FeatureGates != nil {
+		out = append(out, kutil.FeatureGatesToCommandLineParameter(k.values.FeatureGates))
+	}
+
+	out = append(out, "--insecure-port=0")
+	out = append(out, "--kubelet-preferred-address-types=InternalIP,Hostname,ExternalIP")
```

Comment:
The restart will anyways happen because some annotations that weren't there before will now be injected due to the `MakeImmutable` call. Right, it should not be a problem.